ComputeDataDrivenFraction.py

```diff
@@ -0,0 +1,111 @@
+                    help='root file with efficiency and acceptance')
+parser.add_argument('fracFileName', metavar='text', default='fracFile.root',
+                    help='root file with prompt (FD) fraction')
+parser.add_argument('outFileName', metavar='text', default='outFile.root', help='root output file name')
+args = parser.parse_args()
+
+# load input file
+effAccFile = TFile.Open(args.effAccFileName)
+hEffAccPrompt = effAccFile.Get('hAccEffPrompt')
+hEffAccFD = effAccFile.Get('hAccEffFD')
+
+fracFile = TFile.Open(args.fracFileName)
+hCorrYieldPrompt = fracFile.Get('hCorrYieldPrompt')
+hCorrYieldFD = fracFile.Get('hCorrYieldFD')
+hCovPromptPrompt = fracFile.Get('hCovPromptPrompt')
+hCovPromptFD = fracFile.Get('hCovPromptFD')
+hCovFDFD = fracFile.Get('hCovFDFD')
+
+# systematic uncertainties --> total taken as sum in quadrature (uncorrelated)
+hPromptFrac = hEffAccPrompt.Clone('hPromptFrac')
+hFDFrac = hEffAccFD.Clone('hFDFrac')
+hPromptFrac.SetTitle(';#it{p}_{T} (GeV/#it{c}); #it{f}_{prompt}')
+hFDFrac.SetTitle(';#it{p}_{T} (GeV/#it{c}); #it{f}_{FD}')
+
+for iPt in range(hEffAccPrompt.GetNbinsX()):
+    ptMin = hEffAccPrompt.GetBinLowEdge(iPt+1)
+    ptMax = ptMin+hEffAccPrompt.GetBinWidth(iPt+1)
+    ptCent = hEffAccPrompt.GetBinCenter(iPt+1)
+    effAccPrompt = hEffAccPrompt.GetBinContent(iPt+1)
+    effAccFD = hEffAccFD.GetBinContent(iPt+1)
+    effAccPromptUnc = hEffAccPrompt.GetBinError(iPt+1)
+    effAccFDUnc = hEffAccFD.GetBinError(iPt+1)
+
+    # ingredients for (prompt or FD) fraction computation
+    corrYieldPrompt = hCorrYieldPrompt.GetBinContent(iPt+1)
+    corrYieldFD = hCorrYieldFD.GetBinContent(iPt+1)
+    covPromptPrompt = hCovPromptPrompt.GetBinContent(iPt+1)
+    covPromptFD = hCovPromptFD.GetBinContent(iPt+1)
+    covFDFD = hCovFDFD.GetBinContent(iPt+1)
+
+    # prompt fraction
+    fPrompt = effAccPrompt * corrYieldPrompt / (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD)
+    defPdeNP = (effAccPrompt * (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD) - effAccPrompt**2
+                * corrYieldPrompt) / (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD)**2
+    defPdeNF = - effAccFD * effAccPrompt * corrYieldPrompt / \
+        (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD)**2
+    fPromptUnc = np.sqrt(defPdeNP**2 * covPromptPrompt + defPdeNF**2 * covFDFD + 2 * defPdeNP * defPdeNF * covPromptFD)
+
+    # feed-down fraction
+    fFD = effAccFD * corrYieldFD / (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD)
+    defFdeNF = (effAccFD * (effAccFD * corrYieldFD + effAccPrompt * corrYieldPrompt) - effAccFD**2
+                * corrYieldFD) / (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD)**2
+    defFdeNP = - effAccFD * effAccPrompt * corrYieldFD / \
+        (effAccPrompt * corrYieldPrompt + effAccFD * corrYieldFD)**2
+    fFDUnc = np.sqrt(defFdeNF**2 * covFDFD + defFdeNP**2 * covPromptPrompt + 2 * defFdeNF * defFdeNP * covPromptFD)
+
```

Comment:
This part is the same as in `ComputeDataDrivenCrossSection.py`. Could you use a utility function to avoid code duplication?

Reply:
Ok! I would go first through your PR https://github.com/DmesonAnalysers/DmesonAnalysis/pull/199 and merge it to avoid conflicts, since you also modified the `AnalysisUtils.py`

Reply:
Done in f3902d7